### PR TITLE
feat!: versioned namespaces

### PR DESCRIPTION
**Pull request by rootulp**
Part of https://github.com/celestiaorg/celestia-app/issues/1434

Breaking b/c adds a field to the `Blob` type.

### CODE REVIEW

**Review by evan-forbes — APPROVED** [review]
yah, LGTM

**Comment by evan-forbes** [comment]
will review again when ready and that last test is passing :slightly_smiling_face:

**Comment by rootulp** [comment]
I'll investigate https://github.com/celestiaorg/celestia-core/actions/runs/4501491369/jobs/7924181438?pr=985#step:6:110

File: types/block.go [hunk 1]
```diff
@@ -1082,10 +1085,16 @@ func BlobFromProto(p *tmproto.Blob) Blob {
 	if p == nil {
 		return Blob{}
 	}
+
+	if p.NamespaceVersion > math.MaxUint8 {
+		return Blob{}
+	}
```
Comment (cmwaters) [inline types/block.go:1091]: Do we want the overflow check for `ShareVersion` as well?

**Review by cmwaters — APPROVED** [review]
LGTM

File: types/block.go (hunk 1 above)
Comment (rootulp) [reply types/block.go:1095]: resolved in https://github.com/celestiaorg/celestia-core/pull/985/commits/809ceb5da86339f8798a32c70f3577619f836e38

**Pull request merged**